internal/controller/challenges/apply_test.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2020 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package challenges
+
+import (
+	"strconv"
+	"sync"
+	"testing"
+
+	fuzz "github.com/google/gofuzz"
+	"github.com/stretchr/testify/assert"
+
+	cmacme "github.com/jetstack/cert-manager/pkg/apis/acme/v1"
+)
+
+func Test_serializeApply(t *testing.T) {
+	const (
+		expReg  = `^{"kind":"Challenge","apiVersion":"acme.cert-manager.io/v1","metadata":{.*},"spec":{.*},"status":{"processing":false,"presented":false}}$`
+		numJobs = 10000
+	)
+
+	var wg sync.WaitGroup
+	jobs := make(chan int)
+
+	wg.Add(numJobs)
+	for i := 0; i < 3; i++ {
+		go func() {
+			for j := range jobs {
+				t.Run("fuzz_"+strconv.Itoa(j), func(t *testing.T) {
+					var challenge cmacme.Challenge
+					fuzz.New().NilChance(0.5).Funcs(
+						func(challenge *cmacme.Challenge, c fuzz.Continue) {
+							if challenge.Spec.Solver.DNS01 != nil && challenge.Spec.Solver.DNS01.Webhook != nil {
+								challenge.Spec.Solver.DNS01.Webhook.Config = nil
```

Comment:
Question: why not test with this field set to an actual value by the fuzzer?

Reply:
Ah, I should have added a comment. `Config` has a type of `v1.JSON` and will panic if set to a value which is not a struct of correctly parse json. It seems silly to setup all the scaffolding to create random JSON just for this one field.

Happy to spend the time to do it if you think it is valuable. Will add a comment in anycase.

Reply:
I've gone ahead and set it to a static JSON value.